dms-infrastructure/src/main/kotlin/team/aliens/dms/persistence/student/StudentPersistenceAdapter.kt

```diff
@@ -156,6 +150,20 @@ class StudentPersistenceAdapter(
             }
     }
 
+    private fun eqStudentRecentPointHistory(): BooleanExpression? {
+        return pointHistoryJpaEntity.studentName.eq(studentJpaEntity.name)
+            .and(eqGcn())
+            .and(pointHistoryJpaEntity.createdAt.eq(
+                select(pointHistoryJpaEntity.createdAt.max())
```

Comment:
p2; max 함수보다 orderBy를 사용해 정렬 후 limit을 거는 것이 더 좋을 것 같아요. 
성능상 큰 차이는 아니지만 실행계획이 한 단계 감소하고, 코드가 간소화됩니다.

Reply:
그렇게 할려고 시도해봤는데 limit은 fetch가 되지 않으면 실행이 안된다고해서... 어쩔 수 없이 저렇게 했습니다.

Reply:
```suggestion
                .and(pointHistoryJpaEntity.createdAt.eq(
                select(pointHistoryJpaEntity.createdAt)
                    .from(pointHistoryJpaEntity)
                    .where(
                        pointHistoryJpaEntity.school.id.eq(schoolJpaEntity.id),
                        pointHistoryJpaEntity.studentName.eq(studentJpaEntity.name),
                        eqGcn()
                    )
                    .orderBy(pointHistoryJpaEntity.createdAt.desc())
                    .limit(1)
            ))
```

Reply:
이게 코드상에서는 있는데 실제 실행되는 쿼리를 보면 limit이 적용안된 상태로 나가서 위에서 말했 듯 같은 학교 A학생과 B학생한테 동시에 벌점을 줬다면 2개이상 선택되기에 예외가 터집니다
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/52336493/220250842-a5af31ca-c09a-4469-8b5e-a9658583d844.png">

Reply:
제가 알기로는 alias를 지정해주면 되는걸로 아는데 안되는 건가요?

```suggestion
                .and(pointHistoryJpaEntity.createdAt.eq(
                Expressions.`as`(
                    select(pointHistoryJpaEntity.createdAt)
                        .from(pointHistoryJpaEntity)
                        .where(
                            pointHistoryJpaEntity.school.id.eq(schoolJpaEntity.id),
                            pointHistoryJpaEntity.studentName.eq(studentJpaEntity.name),
                            eqGcn()
                        )
                        .orderBy(pointHistoryJpaEntity.createdAt.desc())
                        .limit(1), 
                    "recentPointHistory"
                )
            ))
```

Reply:
음 그건 아마 mysql의 경우에 그런거고 에초에 jpql에서 만들때 안되는것 같더라구요 시도해봤는데
<img width="970" alt="image" src="https://user-images.githubusercontent.com/52336493/220264266-c6edbf79-45d4-4e6c-bae3-d85bbe5b4cf8.png">
이렇게 에러뜨고 limit이 적용이 안되어있어요

Reply:
음.. 알겠습니다